### PR TITLE
feat: introduce SearchSG embedded results page

### PR DESCRIPTION
**Pull request by dcshzj**
Refer to the [onboarding guide](https://docs.developer.tech.gov.sg/docs/searchsg-onboarding-guide/customisation-embed?id=step-1-load-the-searchconfigjs-script-for-result-page).

Not doing the search results page yet for local search as that is more complicated + we are prioritising SearchSG integration.

### CODE REVIEW

**Comment by dcshzj** [comment]
!run chromatic

File: src/templates/next/layouts/SearchSG/SearchSG.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  site,
+  page,
+  ScriptComponent = "script",
+}: SearchSGPageSchema) => {
+  const clientId =
+    (site.search && site.search.type === "searchSG" && site.search.clientId) ||
+    ""
+
+  return (
+    <Skeleton site={site} page={page}>
```
Comment (jacobkwan) [inline src/templates/next/layouts/SearchSG/SearchSG.tsx:14]: need to pass in `LinkComponent` to Skeleton too!

File: src/templates/next/layouts/SearchSG/SearchSG.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  title: "Next/Layouts/SearchSG",
+  component: SearchSGLayout,
+  argTypes: {},
+  parameters: {
+    themes: {
+      themeOverride: "Isomer Next",
+    },
+  },
+} as Meta
+const TEST_CLIENT_ID = "7946e346-993e-41c7-bd81-26a3999dc3f4"
```
Comment (jacobkwan) [inline src/templates/next/layouts/SearchSG/SearchSG.stories.tsx:16]: same comment from previous PR about env vars

**Comment by jacobkwan** [comment]
but wow, onboarding does indeed seem very simple 👀

File: src/templates/next/layouts/SearchSG/SearchSG.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    (site.search && site.search.type === "searchSG" && site.search.clientId) ||
+    ""
+
+  return (
+    <Skeleton
+      site={site}
+      page={page}
+      LinkComponent={LinkComponent}
+      HeadComponent={HeadComponent}
+      ScriptComponent={ScriptComponent}
```
Comment (jacobkwan) [inline src/templates/next/layouts/SearchSG/SearchSG.tsx:21]: can help me to add these to the other layouts 😄

**Review by jacobkwan — APPROVED** [review]

File: src/templates/next/layouts/SearchSG/SearchSG.tsx (hunk 3 above)
Comment (dcshzj) [reply src/templates/next/layouts/SearchSG/SearchSG.tsx:21]: Yeah, added in 8ddd83a134333956693f634b012c8370f70adeca.

**Pull request merged**